test/class_test.js

```diff
@@ -1131,4 +1131,65 @@ describe('classes', () => {
       o = b.c(1)
     `, 9);
   });
+
+  it('handles a complex anonymous class in an expression context', () => {
+    check(`
+      A = class
+        b: c
+    `, `
+      const A = __initClass__(class {
+        static initClass() {
+          this.prototype.b = c;
+        }
+      });
+      var __initClass__ = function(c) {
```

Comment:
This will fail since this helper won't be set yet. If we use the backtick approach for the helper we can use a function declaration which will be hoisted.